cbmc/proofs/mqtt_state.c

```diff
@@ -489,6 +489,8 @@ bool valid_IotMqttPublishInfo( const IotMqttPublishInfo_t *pInfo )
   return
     pInfo &&
 
+    (pInfo->qos == IOT_MQTT_QOS_0 || pInfo->qos == IOT_MQTT_QOS_1 || pInfo->qos == IOT_MQTT_QOS_2) &&
```

Comment:
@gordonwang0 can I assume this for all `IotMqttPublishInfo_t ` objects. According to the documentation, `IOT_MQTT_QOS_2` is not entirely supported, but I expect that all functions that deals with `IotMqttPublishInfo_t ` will be resilient to this.

Reply:
Yes, that is correct.